### PR TITLE
38 implement putting to tile yard

### PR DESCRIPTION
**Pull request by Structed**

### CODE REVIEW

**Comment by Structed** [comment]
Had to move the Spare Yard to the main Game Scene. Town is a TileMap and does not allow for proper Control layouting.

File: src/alcaramba/Drawable/Town/SpareTiles.gd [hunk 1]
```diff
@@ -10,24 +10,39 @@ var tile_scene = preload("res://Drawable/Card/TileCardDrawable.tscn")
 func _display_tiles():
```
Comment (zientist) [inline src/alcaramba/Drawable/Town/SpareTiles.gd:10]: move spareTiles.gd to Game folder

File: src/alcaramba/Drawable/Town/SpareTiles.gd [hunk 2]
```diff
@@ -10,24 +10,39 @@ var tile_scene = preload("res://Drawable/Card/TileCardDrawable.tscn")
...
 	# delete already displayed tiles
 	for n in self.get_children():
-		n.queue_free()
+		# Leave other controls there
+		if n is TileCard:
```
Comment (zientist) [inline src/alcaramba/Drawable/Town/SpareTiles.gd:14]: Needs to be TileCardDrawable

File: src/alcaramba/Scripts/Player.gd [hunk 3]
```diff
@@ -20,3 +23,37 @@ func add_town_tile(tile: TileCard, position: Vector2):
...
+# @param position: Vector2 - The Town position of the tile
+# @param int: Returns `-1` in case of an error. `0` otherwise.
+func transfer_tile_spare_to_town(tile: TileCard, position: Vector2) -> int:
+	# Nothing to do if the card does nto exist in the yard
+	if tile_cards_yard.has_card(tile) == false:
+		return 0
+	
+	if town_tiles.has(position.x) && town_tiles[position.x].has(position.y):
+		
+		var check_tile = town_tiles[position.x][position.y]
```
Comment (zientist) [inline src/alcaramba/Scripts/Player.gd:41]: use vector as dictionary key

File: src/alcaramba/Scripts/Player.gd [hunk 4]
```diff
@@ -20,3 +23,37 @@ func add_town_tile(tile: TileCard, position: Vector2):
...
+		if check_tile is int:
+			push_error("There is already a tile at position %d|%d" % [position.x, position.y])
```
Comment (zientist) [inline src/alcaramba/Scripts/Player.gd:43]: dictionary key

**Comment by zientist** [comment]
Tile is not removed from market when buying and putting it on spare

**Pull request merged**